frontend/src/lib/components/icon/Icon.svelte

```diff
@@ -0,0 +1,108 @@
+<script lang="ts">
+  import {_} from 'svelte-i18n';
+  import type {IconProps} from './Icon.type';
+  import {ICONS} from './icons';
+
+  type $$Props = IconProps;
+  export let name: $$Props['name'];
+  export let size: $$Props['size'] = 'md';
+  export let color: $$Props['color'] = 'current';
+
+  // Validate name, split path and get aria-label
+  // We need this part-wise approach because of Vite's dynamic import limitations
+  // https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#limitations
+  let label: string;
+  let folder: string;
+  let filename: string;
+
+  if (name in ICONS) {
+    label = $_(`icons.${name}`);
+    [folder, filename] = ICONS[name].split('/');
+  } else {
+    label = name;
+    folder = 'custom';
+    filename = 'missing_icon';
+  }
+
+  // Create class names
+  let classes = 'vaa-icon inline';
+
+  // Predefined sizes
+  switch (size) {
+    case 'sm':
+      classes += ' h-16 w-16';
+      break;
+    case 'lg':
+      classes += ' h-32 w-32';
+      break;
+    default:
+      classes += ' h-24 w-24';
+  }
+
+  // Fill colors
+  // NB. We need to spell these out because otherwise Tailwind won't compile them
+  const FILLS = {
+    current: 'fill-current',
+    primary: 'fill-primary',
+    secondary: 'fill-secondary',
+    accent: 'fill-accent',
+    neutral: 'fill-neutral',
+    'base-100': 'fill-base-100',
+    'base-200': 'fill-base-200',
+    'base-300': 'fill-base-300',
+    info: 'fill-info',
+    success: 'fill-success',
+    warning: 'fill-warning',
+    error: 'fill-error',
+    'base-content': 'fill-base-content',
+    'primary-content': 'fill-primary-content',
+    'secondary-content': 'fill-secondary-content',
+    'accent-content': 'fill-accent-content',
+    'info-content': 'fill-info-content',
+    'success-content': 'fill-success-content',
+    'warning-content': 'fill-warning-content',
+    'error-content': 'fill-error-content'
+  };
+
+  if (color != null && color in FILLS) {
+    classes += ` ${FILLS[color]}`;
```

Comment:
We could simply receive the colour with the following format `fill-{COLOR}` as suggested in the daisyUI docs https://daisyui.com/docs/colors/#-3.

Reply:
I think the docs refer in that case only to the logic of using the classes, not actual code. See: https://tailwindcss.com/docs/content-configuration#dynamic-class-names

The `fill-XXX` classes don't work, if you first remove them from the component (completely, not just commenting them out) and rebuild the app from scratch (I ran `yarn dev:down; yarn dev`), bc the classes might be cached.